src/tests/class_tests/openms/source/Colorizer_test.cpp

```diff
@@ -0,0 +1,287 @@
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: $
+// $Authors: Moritz Berger, Tetana Krymovska$
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+#include <OpenMS/CONCEPT/Colorizer.h>
+///////////////////////////
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <fstream>
+
+using namespace OpenMS;
+using namespace std;
+
+/* TO-DO 2
+- getReset testable? Why is output always 1?
+- UPDATE OpenMS Version (consult) and Colorizer version
+- Test background color change test case - DEFINE functionality first
+- ConsoleUtils Erweiterung
+*/
+
+template <typename T>
+//convert any input to string
+string convertToString ( T var_input )
+{
+    ostringstream ss;
+    ss << var_input;
+    return ss.str();
+}
+
+START_TEST(Colorizer(),"$Id$")
+
+ //Test variables
+char test_char = 'a';
+int test_int = 15;
+float test_float = 2094.5892;
+string test_string = " !#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+
+//ANSI codes
+
+#ifdef OPENMS_WINDOWSPLATFORM
+
+    string const blackANSI        = "";
+    string const redANSI          = "";
+    string const greenANSI        = "";
+    string const yellowANSI       = "";
+    string const blueANSI         = "";
+    string const magentaANSI      = "";
+    string const cyanANSI         = "";
+    string const whiteANSI        = "";
+    string const resetColorANSI   = "";
+
+#elif defined(__linux__) || defined(__OSX__)
+
+    string const blackANSI        = "\e[30m";
+    string const redANSI          = "\e[31m";
+    string const greenANSI        = "\e[32m";
+    string const yellowANSI       = "\e[33m";
+    string const blueANSI         = "\e[34m";
+    string const magentaANSI      = "\e[35m";
+    string const cyanANSI         = "\e[36m";;
+    string const whiteANSI        = "\e[37m";
+    string const resetColorANSI   = "\e[0m";
+
+#endif
+
+START_SECTION(Colorizer::colorStream(ostream& stream) const)
```

Comment:
since `Colorizer::colorStream_` is protected now, you cannot test for it anymore directly.
But: in the Colorizer_test.cpp, you can create a new derived class, which exposes the protected functions and then write tests for the new Class.